### ISSUE TITLE
Need capability to run multiple forecast segments

### ISSUE THREAD

**Issue by WalterKolczynski-NOAA** [NOAA-EMC/global-workflow#2274]
### What new functionality do you need?

Longer forecasts (such as those for GEFS S2S and SFS) are too long to run in a single job within wallclock limits and/or fit in different, heterogenous chunks in operations.

Depends on #2273
Depends on #2674

### What are the requirements for the new functionality?

The ability to run a forecast in multiple segments, which each segment continuing from the end of the previous one. At least two are needed, but allowing an arbitrary number is best (and may be necessary for SFS).

Stretch scope: allow different resources for each segment. This can be handled as a separate issue unless it makes sense to incorporate into the design here.

### Acceptance Criteria

- [ ] Ability to run a forecast in at least two segments.
- [ ] A new or updated CI test that runs the forecast in segments.
